packages/smooth_app/lib/pages/user_management/sign_up_page.dart

```diff
@@ -0,0 +1,311 @@
+              validator: (String? value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter a password';
+                }
+                if (!UserManagementHelper.isPasswordValid(value)) {
+                  return 'Please enter a valid password (at least 6 characters)';
+                }
+                return null;
+              },
+            ),
+            const SizedBox(height: space),
+            SmoothTextFormField(
+              type: TextFieldTypes.PASSWORD,
+              controller: _password2Controller,
+              hintText: 'Confirm Password',
+              prefixIcon: const Icon(Icons.vpn_key),
+              autofillHints: const <String>[
+                AutofillHints.password,
+              ],
+              validator: (String? value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please confirm the password';
+                }
+                if (value != _password1Controller.text) {
+                  return 'The passwords should be identical';
+                }
+                return null;
+              },
+            ),
+            const SizedBox(height: space),
+            Text('Username cannot contains spaces, caps or special characters'),
+            const SizedBox(height: space),
+            // careful with CheckboxListTile and hyperlinks
+            // cf. https://github.com/flutter/flutter/issues/31437
+            ListTile(
+              leading: Checkbox(
+                value: _agree,
+                onChanged: (final bool? value) {
+                  if (value != null) {
+                    setState(() => _agree = value);
+                  }
+                },
+              ),
+              title: RichText(
+                text: TextSpan(
+                  children: <InlineSpan>[
+                    TextSpan(
+                      text: 'I agree to the Open Food Facts ',
+                    ),
+                    TextSpan(
+                      style: TextStyle(
+                        color: Colors.blue,
+                        decoration: TextDecoration.underline,
+                      ),
+                      text: 'terms of use and contribution',
+                      recognizer: TapGestureRecognizer()
+                        ..onTap = () async {
+                          final String url =
+                              'https://github.com/flutter/gallery/';
+                          if (await canLaunch(url)) {
+                            await launch(
+                              url,
+                              forceSafariVC: false,
+                            );
+                          }
+                        },
+                    ),
+                  ],
+                ),
+              ),
+              subtitle: !_disagreed
+                  ? null
+                  : Text(
+                      'You have to agree my friend!',
```

Comment:
```suggestion
                      'You have to agree if you'd like to get an account. Note that you can still use the app and contribute anonymously.',
```
This hints at a legal paradox, the user is already implicitely agreing to the terms which are also on the playstore listing.